lib/util/is-typescript.js

```diff
@@ -11,6 +11,8 @@ const typescriptExtensions = [".ts", ".tsx", ".cts", ".mts"]
  * @returns {boolean}
  */
 module.exports = function isTypescript(context) {
-    const sourceFileExt = path.extname(context.getPhysicalFilename())
+    const sourceFileExt = path.extname(
+        context.getPhysicalFilename() ?? context.physicalFilename
```

Comment:
**issue (blocking)**: I think this will break unless we have this the other way around, as in eslint v9 `context.getPhysicalFilename()` will be removed
```suggestion
        context.physicalFilename ?? context.getPhysicalFilename()
```

Reply:
ops, I just missed it.